src/__tests__/utils/ProviderSupportStore.test.ts

```diff
@@ -0,0 +1,362 @@
+import { describe, it } from '@jest/globals'
+
+import { CryptoKey, PaymentKey, ProviderSupportStore } from '../../plugins/gui/providers/ProviderSupportStore'
+
+describe('ProviderSupportStore', () => {
+  const generalStore = makeGeneralStoreFixture()
+
+  it('toJsonObject, toJson, fromJsonObject, fromJson', () => {
+    const obj = generalStore.toJsonObject()
+    const json = generalStore.toJson()
+
+    const expectedObj = {
+      buy: true,
+      '*': {
+        US: true,
+        'US:CA': true,
+        UK: true,
+        '*': {
+          'iso:USD': true,
+          'iso:CAD': true,
+          'iso:GBP': true,
+          '*': {
+            ach: true,
+            sepa: true,
+            credit: true,
+            '*': {
+              'ethereum:null': true,
+              'ethereum:USDC': true,
+              'bitcoin:null': true
+            }
+          }
+        }
+      }
+    }
+    expect(obj).toEqual(expectedObj)
+
+    const expectedJson = JSON.stringify(expectedObj)
+    expect(json).toBe(expectedJson)
+
+    generalStore.fromJsonObject(obj)
+    expect(generalStore.toJsonObject()).toEqual(expectedObj)
+
+    generalStore.fromJson(json)
+    expect(generalStore.toJson()).toEqual(expectedJson)
+  })
+
+  it('toJsonObject', () => {
+    const obj = generalStore.toJsonObject()
+
+    expect(obj).toEqual({
+      buy: true,
+      '*': {
+        US: true,
+        'US:CA': true,
+        UK: true,
+        '*': {
+          'iso:USD': true,
+          'iso:CAD': true,
+          'iso:GBP': true,
+          '*': {
+            ach: true,
+            sepa: true,
+            credit: true,
+            '*': {
+              'ethereum:null': true,
+              'ethereum:USDC': true,
+              'bitcoin:null': true
+            }
+          }
+        }
```

Comment:
Better to just not allow false since only `fromJson` would allow that to happen, not the normal api. see comments below

Reply:
The motivation to having `false`: ability to disable entries in the info server temporarily. But this is a moot case since there is no way to disable an entire object structure without removing it. So this "feature" is only applicable to leaf nodes. I no longer think the benefit for this feature is worth the tradeoff in ambiguity, so I'll remove `false` from the data type.